### DISCUSSION TITLE
devpi install --venv - background research and solution

### DISCUSSION THREAD

**Discussion by nyue** [devpi/devpi#926] [Q&A]
I recently upgraded to devpi-client 6.x and my existing workflow of using devpi install with the venv option failed.

I am hoping to learn the reason behind the removal of the --venv option from install and what is the correct way to achieve the same effect as per the <6.0 version of devpi-client

Thank you

**Comment by fschulze** [comment]
The option still exists, but it requires an existing virtualenv. It was too easy to accidentally create a virtualenv when making a typo in the path. You first have to use ``virtualenv`` to create one, then you can use ``devpi install --venv ...`` to install into it.

I hope this explanation helps you.
